specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2024-09-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,3168 @@
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ApiKeyListResult": {
+      "description": "The result of a request to list API keys.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The collection value.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiKey"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next set of paged results.",
+          "type": "string"
+        }
+      }
+    },
+    "ApiKey": {
+      "description": "An API key used for authenticating with a configuration store endpoint.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The key ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "A name for the key describing its usage.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The value of the key that is used for authentication purposes.",
+          "type": "string",
+          "readOnly": true
+        },
+        "connectionString": {
+          "description": "A connection string that can be used by supporting clients for authentication.",
+          "type": "string",
+          "readOnly": true
+        },
+        "lastModified": {
+          "format": "date-time",
+          "description": "The last time any of the key's properties were modified.",
+          "type": "string",
+          "readOnly": true
+        },
+        "readOnly": {
+          "description": "Whether this key can only be used for read operations.",
+          "type": "boolean",
+          "readOnly": true
+        }
+      }
+    },
+    "RegenerateKeyParameters": {
+      "description": "The parameters used to regenerate an API key.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The id of the key to regenerate.",
+          "type": "string"
+        }
+      }
+    },
+    "GenerateSasTokenParameters": {
+      "description": "Parameters used for generating SAS token.",
+      "required": [
+        "resourceType",
+        "exp",
+        "sasKind"
+      ],
+      "type": "object",
+      "properties": {
+        "resourceType": {
+          "description": "The data plane resource type that the SAS token is authorized to access.",
+          "type": "string",
+          "enum": [
+            "Kv",
+            "Snapshot"
+          ],
+          "x-ms-enum": {
+            "name": "resourceType",
+            "modelAsString": true
+          }
+        },
+        "signedParameters": {
+          "description": "The parameters to define the data plane resource scope.",
+          "$ref": "#/definitions/SignedParameters"
+        },
+        "exp":{
```

Comment:
I would expect full names for ARM API.

Reply:
Are you suggesting we accept full names in request body and abbreviate as needed while generating the token?

Reply:
Yes.

Reply:
Okay, so we can change these three request paramters:
- exp => expires
- cch => cache_control_max_age
- sas_kind => kind

Reply:
Got it

Reply:
How about expiration instead of expires?

Reply:
We use `expires` for snapshots: 
https://github.com/Azure/azure-rest-api-specs/blob/0117e474a03d0b5b4cdd3b713105ae17196c18ce/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json#L2978

Reply:
Ah, okay thank you. I thought we must have a precedent. expires it is then.

Reply:
For resourceType value, should we keep "kv" or update to "keyValue"?

Reply:
Keep `kv`. The concern I was raising was purely over property names.